src/Spatial/Euclidean/Quaternion.cs

```diff
@@ -310,15 +334,28 @@ public Quaternion RotateUnitQuaternion(Quaternion unitQuaternion)
         /// </summary>
         public static Quaternion operator *(Quaternion q1, double d)
         {
-            return q1.Multiply(d);
+            return new Quaternion(q1.Real * d, q1.ImagX * d, q1.ImagY * d, q1.ImagZ * d);
         }
 
         /// <summary>
-        /// Divide a quaternion by a quaternion.
+        /// Multiply a floating point number with a quaternion.
         /// </summary>
-        public static Quaternion operator /(Quaternion q1, Quaternion q2)
+        public static Quaternion operator *(double d, Quaternion q1)
+        {
+            return new Quaternion(q1.Real * d, q1.ImagX * d, q1.ImagY * d, q1.ImagZ * d);
+        }
+        /// <summary>
+        /// Divide a quaternion by a quaternion.
+        /// </summary> 
+        public static Quaternion operator /(Quaternion q, Quaternion r)
         {
-            return q1.Divide(q2);
+            double normSquared = r.NormSquared;
+            var t0 = (r._w * q._w - r._x * q._x - r._y * q._y + r._z * q._z) / normSquared;
```

Comment:
Formatting: we have used no space for a*b and space for a + b. Reads nicely imo.